x-pack/test/search_sessions_integration/tests/apps/discover/async_search.ts

```diff
@@ -60,6 +60,7 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
       url = await browser.getCurrentUrl();
       expect(url).to.contain('searchSessionId');
       await PageObjects.header.waitUntilLoadingHasFinished();
+      // Note this currently fails, for some reason the fakeSearchSessionId is not restored
```

Comment:
dear @Dosant, could you help here, for some reason this test is failing, when the back button changes the URL the searchSessionId doesn't change, which caused this test to fail

Reply:
I think the bug appeared because this `fetch$` is being extensively subscribed to and unsubscribed from 
https://github.com/kertal/kibana/blob/f2998cb8b4ea6e76b2e0524a59b2ffd128c6d529/src/plugins/discover/public/application/apps/main/services/use_saved_search.ts#L210. I think before it was subscribed only during angular controller initialize. You can put a console.log when we subscribe to it and see how often it is called. 

If I read into it right, then these subscribe/unsubscribes are causing weirdness with how `searchSessionManager.newSearchSessionIdFromURL$` is implemented where it is remembering initial query param value and skips emitting it on subsequent subscribes. 
https://github.com/kertal/kibana/blob/f2998cb8b4ea6e76b2e0524a59b2ffd128c6d529/src/plugins/kibana_utils/public/history/history_observable.ts#L40

This probably could be changed on that observable level, but I actually think it would be better to optimize those hooks to make sure they don't do excessive subscribes/unsubscribes because IMO such could cause other bugs and this also redundant work from CPU ticks perspective.

Reply:
for sure, should just subscribe / unsubscribe once! thanks for catching!

Reply:
thx @Dosant, all tests work. btw I would appreciate a search session / autorefresh focused review from your side, also If you have Observable hints, let me hear 😄